### PR TITLE
fix: add local definition of isPrivateIp

### PR DESCRIPTION
**Pull request by wemeetagain**
## Description

Use local defintion of isPrivateIp instead of `private-ip` package
This should reduce bundle size and remove another commonjs sub-dependency (`ipaddr.js`)

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if necessary (this includes comments as well)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Review by maschad — CHANGES_REQUESTED** [review]
You should verify  that this decreases the [bundle size](https://github.com/libp2p/js-libp2p/blob/cce373ff82a959ba4cacc74a8d980109264c1705/packages/kad-dht/.aegir.js#L5C27-L5C27) and update the limits as well for each package so that this change is worthwhile.

File: packages/utils/test/private-ip.spec.ts [hunk 1]
```diff
@@ -0,0 +1,203 @@
+/* eslint-env mocha */
```
[the remark this answers is not in the document]
Comment (wemeetagain) [reply packages/utils/test/private-ip.spec.ts:1]: yes, also the source is a modified version of https://github.com/frenchbread/private-ip/blob/master/src/index.ts

**Comment by wemeetagain** [comment]
> You should verify that this decreases the bundle size

Its guaranteed to be smaller because its the same code, but with two fewer dependencies.

**Comment by wemeetagain** [comment]
@achingbrain ready for re-review

**Pull request merged**